### DISCUSSION TITLE
Producto

### DISCUSSION THREAD

**Discussion by UO250825** [Arquisoft/dede_es5a#20] [General]
¿Qué producto/s se va/n a vender en la tienda?¿Qué atributos tiene un producto?

**Comment by AlonsoA-hub** [comment]
Yo creo que podríamos hacer una tienda de calcetines, que hay diseños interesantes para usar.
Como campos podríamos poner:
- ID
- Nombre
- Precio
- Tipo (Cortos, tobilleros, largo, etc.)
- Marca
- Tamaño
- Color
![image](https://user-images.githubusercontent.com/60401630/154845266-d863a84b-13df-47f0-b293-ba9879d0452d.png)

**Reply by UO250825** [reply]
Quizas añadir un campo stock/disponibilidad. Me surgen dudas sobre cómo tratar el campo tipo , hay que definir qué tipos hay. Y si unos calcetines del mismo modelo pero diferente color son el mismo u otro producto con otro stock.

**Reply by SofiaYYQ** [reply]
Sí, me parece bien lo que dijo Gaspar @UO250825. 
- Yo creo que es más usable tener los calcetines del mismo modelo agrupados que tenerlos repetidos con distintos colores. Aunque, no sé si serán fácil de implementar posteriormente.
- Después, creo que es mejor talla en vez de tamaño 🤔 .
- Un campo de descripción quizás podría venir bien.

**Reply by UO250825** [reply]
Estoy de acuerdo con @SofiaYYQ:
- Talla en vez de tamaño
- Campo descripción

**Reply by diegoglezsu** [reply]
Una duda que me surge es ¿cómo guardaríamos el campo tipo?, es decir, por ejemplo tipo "largo" o "Largo", podrían generar problemas si se guarda como una cadena de texto... Lo mismo para los colores... 
Quizás podríamos separarlo y que en la base de datos tengamos un conjunto de colores con su nombre y su RGB (quizá tener el rgb pueda ser interesante para evitar hardcodear colores en la interfaz, si por ejemplo se usa un filtro por colores). Añadir un nuevo color sería simplemente añadirlo a la base de datos sin necesidad de tocar la interfaz ya que esta consultará todos los colores disponibles en la base de datos.
Es un poco lo que decía @SofiaYYQ de tener los colores agrupados. 🤔 ¿Qué pensais?

**Reply by diegoglezsu** [reply]
A todo estos, Los atributos los veo bien!! 😄

**Reply by AlonsoA-hub** [reply]
Varias cosas:

- Talla en vez de tamaño y campo descripción buena idea.
- MongoDB se puede poner 'case-insensitive' así que el tema de las mayúsculas no importaría.
- Con color me refería al típico filtro de las paginas de ropa para buscar prendas que contengan el color filtrado, en la BD, los productos pueden tener el campo color, siendo este una lista de colores. De esta forma, miramos los productos que contengan el color seleccionado por el usuario y los devolvemos.
![image](https://user-images.githubusercontent.com/60401630/155388008-605ff5fa-6e33-4631-aa5b-68cb42bec883.png)

**Reply by diegoglezsu** [reply]
Si @AlonsoA-hub a eso me refería vaya, que el color debería estar en la base de datos en una lista de colores por ejemplo. Genial! Si mongoDB ofrece case insensitive genial entonces, no hay problema.
